### PR TITLE
Builder for stabchain that allows us to select the implementation we want

### PR DESCRIPTION
**Pull request by WizardOfMenlo**
So this was my idea for how to handle it

### CODE REVIEW

**Comment by EwanGilligan** [comment]
This is what I had in mind for the builders, although better than I was playing around with. I don't know if there is anything you think should be changed about it, but it looks good to me and would do everything we need it to. I felt having something like this will make it easier to try out and benchmark different algorithms for constructing stabilizer chains.

File: src/group/stabchain/builder/ift.rs [hunk 1]
```diff
@@ -135,13 +136,18 @@ impl<T: MovedPointSelector> StabchainBuilderIFT<T> {
...
         self.chain[self.current_pos] = record;
     }
+}
 
-    pub(super) fn extend(&mut self, p: Permutation) {
+impl<M> super::Builder for StabchainBuilderIFT<M>
+where
+    M: MovedPointSelector,
+{
+    fn extend(&mut self, p: Permutation) {
```
Comment (EwanGilligan) [inline src/group/stabchain/builder/ift.rs:145]: My only comment would be that not all algorithms work with an extend method, so this probably need to be more general. For example with the random generation, there are two separate stages for a strong generating set construction, and then a test for the strong generation set.

Comment (WizardOfMenlo) [reply src/group/stabchain/builder/ift.rs:145]: No worries, I'll refactor with a set_generators method that should be generic enough for any of our needs

Comment (WizardOfMenlo) [reply src/group/stabchain/builder/ift.rs]: I've pushed the fix, let me know what you think

Comment (EwanGilligan) [reply src/group/stabchain/builder/ift.rs]: Yeah looks good to me now. I've approved

**Pull request merged**